# HG changeset patch
# User Rik <rik@octave.org>
# Date 1547668569 28800
#      Wed Jan 16 11:56:09 2019 -0800
# Node ID 8bed6950dddef1a1c35cead76326223e5b4bc105
# Parent  a595c22464e4bb39cd6a8643bb9f1de100aa8580
light.m: Change %!demo code #5 to produce comparable results under Matlab (bug #55841).

* light.m: Use linspace rather than range operator to create a range for
meshgrid with one more data point (12 vs. 11) which produces better visuals.

diff --git a/scripts/plot/draw/light.m b/scripts/plot/draw/light.m
--- a/scripts/plot/draw/light.m
+++ b/scripts/plot/draw/light.m
@@ -193,17 +193,18 @@ endfunction
 %!  h_light1 = light ("Color", "r");
 %!  h_light2 = light ("Position", [0 1 1], "Color", "b");
 %!  h_light3 = light ("Position", [-1 -1 2], "Color", "g");
 
 %!demo
 %! ## Diffuse and specular reflection
 %! clf;
 %! h_axes = axes ();
-%! [x,y,z] = meshgrid (-.2:0.04:.2, -.2:0.04:.2, -.2:0.04:.2);
+%! rng = linspace (-0.2, +0.2, 12);
+%! [x,y,z] = meshgrid (rng);
 %! val = (x.^2 + y.^2 + z.^2);
 %!
 %! fv = isosurface (x, y, z, val, .039);
 %! h_patch = patch (fv, "FaceColor", "r", "EdgeColor", "none", ...
 %!                      "FaceLighting", "Gouraud");
 %! set (h_patch, "DiffuseStrength", 0, "SpecularStrength", 0);
 %! isonormals (x, y, z, val, h_patch);
 %! fv = isosurface (x+.5, y, z, val, .039);
